swingset/src/main/java/com/nqadmin/swingset/SSDBComboBox.java

```diff
@@ -1484,16 +1481,29 @@ public void setSeperator(final String _separator) {
 	 */
 	public boolean updateOption(final Long _mapping, final String _option) {
 
+		// 2021-02-28: @errael patched this method to deal with inconsistent
+		// updating of combo editor. See https://github.com/bpangburn/swingset/issues/85
+		
 		boolean result = false;
 
 		try (Model.Remodel remodel = optionModel.getRemodel()) {
 			final int index = remodel.getMappings().indexOf(_mapping);
 			if (index >= 0) {
+				boolean isSelectedItem = Objects.equals(_mapping, getSelectedMapping());
 				remodel.setOption(index, _option);
 				result = true;
+				// TODO: Possible that this block and isSelectedItem variable
+				// may be eliminated if a bug is discovered/fixed in GlazedLists.
```

Comment:
Changing what's in the ComboEditor, which may be done indirectly when modifying the current item, might change the currently selected item when STRICT. So something is needed to insure that the selected mapping before the change is the selected mapping after the change. Otherwise the first item in the list becomes selected (as has been seen, and I guess that's what kicks the navigator?)

Notice that the bug https://github.com/glazedlists/glazedlists/issues/702 affects subsequent programmatic changes to the underlying EventList, not the first one.

Reply:
@errael are you saying that we would want to retain the `if (isSelectedItem) {` check regardless of whether https://github.com/glazedlists/glazedlists/issues/702 turns out to be a bug and is fixed? In that case we should just eliminate the TODO and possibly replace with a comment similar to what you wrote above?

Reply:
> are you saying that we would want to retain the `if (isSelectedItem) {` check

Yes, that's what I meant. My assumption is that the mapping that was selected before `updateOption` is called should be the mapping after the call. Typically, if the selected item is changed, glazed will change the selection.

>  eliminate the TODO

That works. The comment that's there may need adjustment/clarification.

If, `updateOption` doesn't need to maintain the selected mapping, then the `if` could be removed after the glazed bug is fixed. I don't recall exactly, but I think the `setSelectedItem(item)` is what avoid the glazed bug on the *next* change.